src/ResponseBuilder.cpp

```diff
@@ -112,9 +112,11 @@ void ResponseBuilder::appendHeaders(const HTTPRequest& request)
 	m_responseHeader << "Date: " << webutils::getGMTString(time(0), "%a, %d %b %Y %H:%M:%S GMT") << "\r\n";
 	// Location
 	std::map<std::string, std::string>::const_iterator iter = request.headers.find("location");
-	if (iter != request.headers.end()) {
+	if (iter != request.headers.end() && request.httpStatus != StatusCreated) {
 		m_responseHeader << "Location: " << iter->second << "\r\n";
 	}
+	if (request.httpStatus == StatusCreated)
+		m_responseHeader << "Location: " << request.uri.path << "\r\n";
```

Comment:
I think it would be cleaner to add the location header in ResponseBodyHandler, after file was successfully created. That way we could just check here if it exists and if so we also send it

Reply:
Good point, changed